Consul.Test/AgentTest.cs

```diff
@@ -515,6 +515,33 @@ public async Task Agent_Monitor()
             }
         }
 
+        [Fact]
+        public async Task Agent_MonitorJSON()
+        {
+            using (var logs = await _client.Agent.MonitorJSON(LogLevel.Trace))
+            {
+                var counter = 0;
+                var logsTask = Task.Run(async () =>
+                {
+                    foreach (var line in logs)
+                    {
+                        // Make a request each time so we get more logs
+                        await _client.Agent.Self();
+                        Assert.StartsWith("{", await line);
```

Comment:
I think that trying to deserialize the line is better method of checking whether it is a json object, e.g.:
```Assert.NotNull(JsonConvert.DeserializeObject(await line));```

Reply:
Should we start using `System.Json.Text` instead? I saw that there are plans for a migration..

Reply:
> Should we start using `System.Json.Text` instead? I saw that there are plans for a migration.

Yeah, we have plans to start using `System.Json.Text` for the library. In the test, it doesn't really matter what we will use though, so it is up to you.

Reply:
Hmm, I see that the usage of `Newtonsoft.Json` in the test is problematic for the net461 target (We use ILMerge for net461). So `System.Json.Text` seem to be a better choice.

Reply:
Hmm, I don't get why the tests are failing for `JsonDeserialize`..

Reply:
Ah, I see.. It is failing for versions below v1.7..

Reply:
> Ah, I see.. It is failing for Consul versions below v1.7..

Yes, we have a mechanism to run tests conditionally depending on the agent version. See ACLTests.cs